DataServer/portAllocator.py

```diff
@@ -1,19 +1,41 @@
+import socket
+
+
 class PortAllocator:
     """This class provides a method to allocate the next available port in the
     given block of ports.
     """
+
     def __init__(self, block_start: int, block_end: int) -> None:
         """Initializes the allocator with the specified block of ports
-
         Args:
             block_start (int): Starting number of block of ports, inclusive
             block_end (int): Ending number of block of ports, inclusive
         """
+
+        self.__s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
         self.__start = block_start
         self.__end = block_end
 
+        self.__portDict = {}
+        self.__hostname = self.__s.gethostname()
+
+        self.__ip = self.__s.gethostbyname(self.__hostname)
+
         # TODO reimplement this
-        self.__current = self.__start
+        
+    def isOpen(self, ip, port):
+        try:
+            self.__s.connect((ip, port))
+            self.__s.shutdown(2)
```

Comment:
This call does not release the underlying socket.  See https://docs.python.org/3/library/socket.html#socket.socket.shutdown and https://docs.python.org/3/library/socket.html#socket.socket.close